MailCore/ViewModel/AvatarViewModel.swift

```diff
@@ -0,0 +1,63 @@
+/*
+ Infomaniak Mail - iOS App
+ Copyright (C) 2022 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import SwiftUI
+
+/// Something that can asynchronously load a contact, and update views accordingly
+@MainActor
+public final class AvatarViewModel: ObservableObject {
+    @Published public var displayablePerson: CommonContact
+
+    /// Something not using the MainActor
+    private struct AvatarLoader {
+        var contactConfiguration: ContactConfiguration
+
+        /// Async get contact method
+        func getContact() async -> CommonContact {
```

Comment:
Let me know if this async code is to your liking @PhilippeWeidmann 

Here I used an internal type `AvatarLoader` to escape the main actor limitation of the view model.